### PR TITLE
Move addons to their own directory.

### PR DESCRIPTION
**Pull request by jeffmendoza**
Addons that have built code or docker images include Grafana and
Servicegraph. These are not a part of Mixer. These are built and
included with the default release of Istio.

### CODE REVIEW

**Comment by ldemailly** [comment]
I expect this would imply a change to the release tar (cc @mattdelco ) and to the documentation / istio.io (cc @geeknoid )

**Comment by jeffmendoza** [comment]
I took a look, and it's not likely. This change ends once the docker images are pushed, they should be pushed in the same way as before. The release tarball and docs care about the install files under `/install/...`, and those don't change. Please confirm I'm making sense, more eyes appreciated.

**Comment by mattdelco** [comment]
The docker images don't end up in the tars so I wouldn't expect any issue there.  The release build does store a tar.gz of each docker image in GCS (mostly for backup and republishing purposes), but that's not based on the source file location (it's just an export-to-file of the local docker registry).

While I've got someone on the line about these docker images, I'd like to know if there'd be any objection if these docker files were eventually renamed to:

Dockerfile.servicegraph
Dockerfile.servicegraph_debug
Dockerfile.grafana

These names would allow the makefile recipes for these images to be more akin to the other docker images, and also help towards enabling parallel builds (by using unique names that won't clobber each other).

**Comment by jeffmendoza** [comment]
>While I've got someone on the line about these docker images, I'd like to know if there'd be any objection if these docker files were eventually renamed to:

Sounds good.

**Comment by mattdelco** [comment]
Also, if this gets in sooner than later then someone can close https://github.com/istio/istio/pull/3079 .

**Comment by ldemailly** [comment]
I didn't really read, my bad (surfing too many notifications) - @mattdelco want to / l g t m this one?

**Comment by mattdelco** [comment]
/lgtm

I've generally been cautious on what I mark since for changes like this I vet based on the appropriateness/correctness of a build change but I don't want the bot to eagerly merge based on it (iow, when it comes to things like renames, adding components, etc., I feel like it's more appropriate for someone with more history/context to give on okay, and I don't want to preempt that with an early build review [nor do I want that review to be perceived as an implicit formal okay of those other aspects] ).

File: addons/grafana/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Custom [Grafana](https://grafana.com/) image
+
+[`start.sh`](start.sh) is the entrypoint, and it runs
+[`import_dashboard.sh`](import_dashboard.sh) in the background then
+starts Grafana. `import_dashboard.sh` POSTs the included dasbboards
```
Comment (douglas-reid) [inline addons/grafana/README.md]: s/dasbboards/dashboards

File: addons/grafana/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+and Prometheus data-source config to Grafana once it starts up.
+
+A few env vars are set in the [Dockerfile](Dockerfile) to configure
+Grafana, and more environment-specific ones are set in the [install
+config](/install/kubernetes/templates/addons/grafana.yaml.tmpl).
```
Comment (douglas-reid) [inline addons/grafana/README.md]: Let's add a section on which dashboards are included with a brief (5 words or less) summary of each one.

File: addons/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Istio Addons
+
+This directory contains components that are not a part of core Istio,
+but are built and included with the Istio release. Files that are to
+be compiled and/or be built into Docker images go here. Install files
+still go under [`/install`](/install).
+
+### [`grafana`](grafana)
+
+Files for custom Grafana docker image. See
```
[the remark this answers is not in the document]
Comment (jeffmendoza) [reply addons/README.md]: Done, the heading links to the directory w/ the README

File: addons/grafana/README.md (hunk 1 above)
Comment (jeffmendoza) [reply addons/grafana/README.md]: Done

File: addons/grafana/README.md (hunk 2 above)
Comment (jeffmendoza) [reply addons/grafana/README.md]: Done

**Comment by jeffmendoza** [comment]
Rebased to re-trigger tests, and they're green for now, yay.

@douglas-reid Comments addressed, ptal, thanks.

**Comment by jeffmendoza** [comment]
@douglas-reid Can I get a retest? Thanks!

**Pull request merged**